### PR TITLE
refactor(view): replace underscore event helpers

### PR DESCRIPTION
**Pull request by paulfalgout**
Related #241

## Linked issue

- Related #241

## What

- Add a private captured-tag `isString` helper and reuse it in method normalization and trigger definitions.
- Resolve callable event/trigger maps through `getValue` and iterate a snapshot of their own enumerable string keys.
- Add focused string-classification and View Events contract coverage and regenerate affected main distribution artifacts.

## Why

View Events needs three narrow behaviors that Marionette can own directly: String-tag classification, literal-key configuration resolution, and own-key map iteration. This removes its complete Underscore import and makes the event delegation contract explicit.

## Agent-development failure addressed

- Gives agents one shared, tested String classifier for the remaining UI/View migrations instead of repeating subtle boxed and cross-realm logic.
- Makes event/trigger map scope, snapshot timing, key order, receiver binding, and error cutoff visible in local code and tests.

## Public behavior contract

- Canonical behavior after this PR: callable event and trigger maps run on the view with zero arguments; their own enumerable string keys are snapshotted in order before values are read; nullish resolved maps are no-ops; handlers bind to the view; trigger definitions preserve String-tag and forwarding behavior.
- Superseded behavior removed, if any: direct Underscore `isString`, `result`, and `each` use in View Events; method normalization now reuses the owned String classifier.
- Compatibility exception and removal condition, if any: Underscore sparse-array and generic array-like dense-index iteration are intentionally not reproduced because `events` and `triggers` are documented maps.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: the size/resource contract passes; direct production Underscore importers drop from 12 to 11. Brotli deltas versus base are ESM +53 B, CJS +55 B, UMD +45 B, and minified +21 B.
- Startup/hot path: one String tag reader is captured at module load; delegation keeps one map resolution and one linear own-key pass.
- Allocations/retention: each event/trigger delegation creates one key snapshot, matching the key-snapshot semantics already used by Underscore object iteration.

## Scope

- Affects View Events event/trigger resolution and shared internal String classification.
- Does not change event delegation order, UI normalization, handler lookup, trigger feature flags, DOM delegation, or other Underscore consumers.

## Deployment Impact

- No website, package, tag, release, or deployment is performed by this PR.
- No consumer redeploy is required beyond adopting a future Marionette package containing this change.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm ci --no-audit --no-fund
npm exec vitest -- run test/unit/utils/is-string.spec.js test/unit/mixins/view-events.spec.js test/unit/common/normalize-methods.spec.js test/unit/view.triggers.spec.js test/unit/view.ui-event-and-triggers.spec.js --reporter=dot
npm run lint:ci
npm run build
npm test -- --reporter=dot
npm run test:source
npm run test:dist
npm run test:fixtures
npm run size
npm run test:performance
npm run performance:timing
git diff --check
```

Results: focused tests passed 52/52; full tests passed 1,464/1,464 across 76 files; source, distribution, packed fixture, lint, build, size/resource, 92 performance-contract, hosted timing, and diff validation passed. Independent exact-head review passed 74/74 focused/integration assertions with no findings.

## Package and browser impact

- Entry points or install fixtures affected: main ESM, CJS, UMD, and minified artifacts are regenerated; packed CJS/ESM/type/Vite fixtures pass.
- Browser contracts affected: none; String-tag classification preserves primitive, boxed, cross-realm, tagged, proxy, revoked-proxy, error, and module-load capture behavior.
- Production/dev/test module-graph impact: View Events stops importing Underscore; other production consumers still retain it.

## Rollback or deprecation

- Revert this single commit and regenerate the main distribution artifacts; no data migration or compatibility bridge is involved.

## Review notes

- Snapshot-first iteration preserves key discovery before any handler/value getter, including own `constructor`, `toString`, and `__proto__`, while excluding inherited pollution.
- Generated formats retain one captured String tag reader and one local own-key iterator.

---
## Summary by cubic
Refactors View Events to replace `underscore`'s `isString`, `result`, and `each` helpers with local equivalents, completing the related work in #241. This removes the mixin's direct `underscore` import and makes the event/trigger delegation contract explicit with no change to public behavior.

- Callable `events` and `triggers` maps run on the view with zero arguments; their own enumerable string keys are snapshotted in order before values are read.
- Nullish resolved maps are no-ops.
- Method normalization reuses the shared string-tag classifier in `utils/is-string`.
- Adds unit coverage for string classification and View Events delegation.
- Regenerates dist artifacts (+21 B to +55 B).

<sup>Written for commit 18d63817fb90e2d000bc307bab90e7dc9fd9ba15. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/282?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of view event and trigger mappings, including callable and boxed-string values.
  * Ensured inherited keys are ignored while valid built-in property names remain supported.
  * Improved behavior for null, undefined, dynamic properties, and property access errors.

* **Tests**
  * Added comprehensive coverage for event mappings and string detection across edge cases.

### CODE REVIEW

**Comment by paulfalgout** [comment]
## Code-review approval — exact head `d5a4e65d`

**Approved.** CI green, +174 B, 17,160 remaining. Production Underscore importers **13 → 12**.

Extracting `isString` into `utils/is-string.js` is a straight lift of the captured-tag reader from #277 with no change to the expression, and `normalize-methods.js` now consumes it rather than defining its own — one fewer place for the tag string to drift.

`eachOwn` matches `_.each`'s object path point for point: `Object.keys` is the same own-enumerable-string set `_.keys` produces, both materialize the key array before iterating so mid-loop mutation can't shift the traversal, and `object == null` short-circuits in both. The callback drops the third `(…, obj)` argument, which neither call site used.

The new coverage lines up with the actual risks rather than restating the diff — the snapshot test, the throwing-getter cutoff, and the nullish-map no-op are the three things this rewrite could plausibly have broken, and "preserving own built-in names" pins that an `events` key like `toString` still delegates.

**One honest note on the strength of this review.** `node_modules` is still empty here, so this is reading rather than running — and `mixins/view-events.js` is the first file in the second half of #241 where that distinction bites. Every prototype-collision defect this session came from files of exactly this shape, and in at least one case (#266) the change read as correct and was correct, while the bug lived in what it newly delivered downstream. I traced the analogous path here — an `events` key of `__proto__` now survives into `normalizeUIString` and then `addEventListener('__proto__', …)`, which is inert rather than fatal — but I'm reasoning about that rather than demonstrating it.

Nothing in the diff suggests a problem. I'd just flag that for `ui.js` and `behaviors.js`, the two remaining files with the same shape, reading is a weaker instrument than it has been for the helper substitutions.

<sub>Reviewed with Claude Code.</sub>

File: test/unit/mixins/view-events.spec.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+describe('view events mixin', function() {
+  it('resolves callable event and trigger maps on the view with no arguments', function() {
+    const eventHandler = this.sinon.stub();
+    const events = this.sinon.stub().returns({ click: eventHandler });
+    const triggers = this.sinon.stub().returns({ submit: 'submitted' });
+    const view = createView({ events, triggers });
+    const delegates = [];
+
+    view._delegateEvents(delegates, {});
+    view._delegateTriggers(delegates, {}, view);
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/mixins/view-events.spec.js]: Not changing this because these focused mixin-unit tests must observe internal key-snapshot and getter-order semantics that the public View surface does not expose; adding a supported public API solely for tests would broaden the framework contract.

File: test/unit/mixins/view-events.spec.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+function createView(overrides = {}) {
+  return {
+    normalizeUIString: sinon.stub().callsFake(key => key),
+    triggerMethod: sinon.stub(),
+    ...ViewEventsMixin,
+    ...overrides
+  };
+}
+
+describe('view events mixin', function() {
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/mixins/view-events.spec.js:12]: Handled in `test/unit/mixins/view-events.spec.js` with per-method `#_delegateEvents` and `#_delegateTriggers` describe blocks and split method-specific cases.

**Comment by paulfalgout** [comment]
**Approval carries** to `18d63817` (the SHA I fetched). Tests-only — `dist/`, `mixins/`, `modules/`, and `utils/` are byte-identical to `d5a4e65d`. CI green.

Splitting the combined test into `#_delegateEvents` and `#_delegateTriggers` blocks is worth more than tidiness here. The two methods share `eachOwn` and `getValue` but not their key-handling, so a failure in the shared traversal previously surfaced as one assertion covering both paths — now it fails in one block and passes in the other, which localizes it immediately.

The nullish split is the better half of the reorganization: the events side asserts a **null** resolved map and the triggers side an **undefined** one. Those take the same branch in `eachOwn` today (`object == null`), so covering one of each rather than the same value twice means a future change that starts distinguishing them gets caught on whichever side it breaks.

<sub>Reviewed with Claude Code.</sub>
